### PR TITLE
test images: Adds CoreDNS to dnsutils and jessie-dnsutils images

### LINKED ISSUE #77935: test images: dnsutils and jessie-dnsutils are using dnsmasq
**Which jobs are failing**:

N/A

**Which test(s) are failing**:

    should support configurable pod resolv.conf

**Since when has it been failing**:

N/A

**Testgrid link**:

N/A

**Reason for failure**:

Since ``dnsmasq`` is a Linux specific binary, the tests which are using it cannot pass on Windows. This could be resolved if ``coredns`` was used instead.

**Anything else we need to know**:

/sig testing
/sig windows

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig windows
/sig testing

**What this PR does / why we need it**:

The ``dnsutils`` and ``jessie-dnsutils`` images are installing ``dnsmasq``, which is required for a few tests checking custom DNS servers and configurations.

``dnsmasq`` is a Linux specific binary. In order for the tests to also pass on Windows, this commit adds CoreDNS to the images, so a later commit will update the tests to use CoreDNS instead of ``dnsmasq``.

**Which issue(s) this PR fixes**:

Fixes: #77935
Needed By: #77862

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by chrisohaver** [comment]
> **Which issue(s) this PR fixes:**
Fixes TBA
Needed By: #77861

Should be: "Needed by #77862"

**Comment by bclau** [comment]
> 
> 
> > **Which issue(s) this PR fixes:**
> > Fixes TBA
> > Needed By: #77861
> 
> Should be: "Needed by #77862"

Oups, you're right. :)

File: test/images/dnsutils/Dockerfile [hunk 1]
```diff
@@ -17,3 +17,6 @@ FROM BASEIMAGE
 CROSS_BUILD_COPY qemu-QEMUARCH-static /usr/bin/
 
 RUN apk add --no-cache bind-tools dnsmasq
+
+ADD https://github.com/coredns/coredns/releases/download/v1.5.0/coredns_1.5.0_linux_BASEARCH.tgz /coredns.tgz
```
Comment (aojea) [inline test/images/dnsutils/Dockerfile]: does this work for arm, mips, ppc64le and other non x86 architectures?

Comment (claudiubelu) [reply test/images/dnsutils/Dockerfile]: Yes, that was the point. as you can see in the URL, there's a ``BASEARCH``, which is then replaced accordingly by the ``image-util.sh`` build script. So, the arm image will have the coredns_1.5.0_linux_arm binary, and so on.

I've created a few containers using those images with different archs and tried executing ``/coredns``:

    ubuntu@ubuntu:~/kubernetes/test/images$ docker exec test-arm /coredns --help
    Usage of /coredns:
      -conf string
            Corefile to load (default "Corefile")
      -cpu string
            CPU cap (default "100%")
      -dns.port string
            Default port (default "53")
      -pidfile string
            Path to write pid file
      -plugins
            List installed plugins
      -quiet
            Quiet mode (no initialization output)
      -version
            Show version
    ubuntu@ubuntu:~/kubernetes/test/images$ docker exec test-arm64 /coredns --help
    Usage of /coredns:
      -conf string
            Corefile to load (default "Corefile")
      -cpu string
            CPU cap (default "100%")
      -dns.port string
            Default port (default "53")
      -pidfile string
            Path to write pid file
      -plugins
            List installed plugins
      -quiet
            Quiet mode (no initialization output)
      -version
            Show version
    ubuntu@ubuntu:~/kubernetes/test/images$ docker exec test-ppc64le /coredns --help
    Usage of /coredns:
      -conf string
            Corefile to load (default "Corefile")
      -cpu string
            CPU cap (default "100%")
      -dns.port string
            Default port (default "53")
      -pidfile string
            Path to write pid file
      -plugins
            List installed plugins
      -quiet
            Quiet mode (no initialization output)
      -version
            Show version

**Pull request merged**